src/main/java/org/sagebionetworks/web/client/FeatureFlagConfig.java

```diff
@@ -3,18 +3,25 @@
 import com.google.gwt.json.client.JSONObject;
 import com.google.gwt.json.client.JSONParser;
 import com.google.gwt.json.client.JSONValue;
-import org.sagebionetworks.web.client.cache.SessionStorage;
+import com.google.inject.Inject;
 import org.sagebionetworks.web.client.cookie.CookieProvider;
 
 public class FeatureFlagConfig {
 
   private final CookieProvider cookieProvider;
-  private final JSONObject config;
+  private JSONObject config;
 
-  public FeatureFlagConfig(String json, CookieProvider cookieProvider) {
-    JSONValue parsed = JSONParser.parseStrict(json);
-    config = parsed.isObject();
+  @Inject
+  public FeatureFlagConfig(CookieProvider cookieProvider) {
```

Comment:
Removed the FeatureFlagConfigProvider and made the FeatureFlagConfig a Singleton (from Guice's dependency injection perspective)

Reply:
Also removed the use of session storage.  The feature flags are initialized when the app is loaded.  One can imagine updating the FeatureFlagConfig with another value, that would disappear as soon as you reload the app.  Maybe this isn't the right behavior, but we don't have this UI right now to validate with users.